internal/weaver/remoteweavelet.go

```diff
@@ -430,10 +430,15 @@ func (w *RemoteWeavelet) UpdateComponents(req *protos.UpdateComponentsRequest) (
 // UpdateRoutingInfo implements the conn.WeaverHandler interface.
 func (w *RemoteWeavelet) UpdateRoutingInfo(req *protos.UpdateRoutingInfoRequest) (reply *protos.UpdateRoutingInfoReply, err error) {
 	defer func() {
-		name := logging.ShortenComponent(req.RoutingInfo.Component)
-		routing := fmt.Sprint(req.RoutingInfo.Replicas)
-		if req.RoutingInfo.Local {
-			routing = "local"
+		name := "?"
```

Comment:
Nice catch.

This is too magical, would the following make sense:

```
func (w *RemoteWeavelet) UpdateRoutingInfo(...) (...) {
  if req.RoutingInfo == nil {
    w.syslogger.Debug("nil RoutingInfo")
    return nil, fmt.Errorf("nil RoutingInfo")
  }
  info := req.RoutingInfo
  defer func() {
    // assume info != nil
    ...
  }()
  ...
```

Reply:
Much better, thanks!